samples/snippets/src/main/java/com/example/spanner/admin/generated/CreateInstanceWithAutoscalingConfigExample.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.spanner.admin.generated;
+
+// [START spanner_create_instance_with_autoscaling_config]
+
+import com.google.cloud.spanner.admin.instance.v1.InstanceAdminClient;
+import com.google.spanner.admin.instance.v1.AutoscalingConfig;
+import com.google.spanner.admin.instance.v1.CreateInstanceRequest;
+import com.google.spanner.admin.instance.v1.Instance;
+import com.google.spanner.admin.instance.v1.InstanceConfigName;
+import com.google.spanner.admin.instance.v1.ProjectName;
+import java.io.IOException;
+import java.util.concurrent.ExecutionException;
+
+class CreateInstanceWithAutoscalingConfigExample {
+
+  static void createInstance() throws IOException {
+    // TODO(developer): Replace these variables before running the sample.
+    String projectId = "my-project";
+    String instanceId = "my-instance";
+    createInstance(projectId, instanceId);
+  }
+
+  static void createInstance(String projectId, String instanceId) throws IOException {
+    InstanceAdminClient instanceAdminClient = InstanceAdminClient.create();
+
+    // Set Instance configuration.
+    String configId = "regional-us-central1";
+    String displayName = "Descriptive name";
+
+    // Create an autoscaling config.
+    AutoscalingConfig autoscalingConfig =
+        AutoscalingConfig.newBuilder()
+            .setAutoscalingLimits(
+                AutoscalingConfig.AutoscalingLimits.newBuilder().setMinNodes(1).setMaxNodes(2))
+            .setAutoscalingTargets(
+                AutoscalingConfig.AutoscalingTargets.newBuilder()
+                    .setHighPriorityCpuUtilizationPercent(65)
+                    .setStorageUtilizationPercent(95))
+            .build();
+    Instance instance =
+        Instance.newBuilder()
+            .setAutoscalingConfig(autoscalingConfig)
```

Comment:
nit: Should we explicitly call out in a comment that we are not setting a node count or processing units here, as we are seting an AutoscalingConfig?

Reply:
This is there is the proto documentation, but I don't see any harm in communicating again over here. Added it here as well.